internal/data.go

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) 2024 Seoyoung Cho.
+
+package internal
+
+import (
+	"database/sql"
+)
+
+type User struct {
```

Comment:
Explain the reason for this change and what else it affects in the code base?

Reply:
This is the first time we're using the struct, so it literally affected nothing. I actually discovered something cool about the sql package were using:
Because go doesn't have null safety, they can't return null for a field that was never set.
So, they have this sql.Null wrapper that has a .Valid boolean attribute that we can use to verify that the data was at least set once, if not we must consider this row in the database malformed.

Reply:
Understood, great work!